From: Ben Hutchings <ben@decadent.org.uk>
Subject: ixgbe: Revert fix for IPv6 GSO type checks

In 2.6.32-39 I cherry-picked commit
8e1e8a4779cb23c1d9f51e9223795e07ec54d77a which affected e1000e, igb,
igbvf and ixgbe.  This was also included in longterm update 2.6.32.47
so I should revert it before applying that.  However, since I have
also backported e1000e, igb and igbvf, any changes to them are
automatically filtered out of longterm updates.  Therefore the change
only needs to be reverted for ixgbe.

Confused?  You won't be after this week's installment of linux-2.6.

diff --git a/drivers/net/ixgbe/ixgbe_main.c b/drivers/net/ixgbe/ixgbe_main.c
index 6810149..a550d37 100644
--- a/drivers/net/ixgbe/ixgbe_main.c
+++ b/drivers/net/ixgbe/ixgbe_main.c
@@ -4881,7 +4881,7 @@ static int ixgbe_tso(struct ixgbe_adapter *adapter,
 			                                         IPPROTO_TCP,
 			                                         0);
 			adapter->hw_tso_ctxt++;
-		} else if (skb_is_gso_v6(skb)) {
+		} else if (skb_shinfo(skb)->gso_type == SKB_GSO_TCPV6) {
 			ipv6_hdr(skb)->payload_len = 0;
 			tcp_hdr(skb)->check =
 			    ~csum_ipv6_magic(&ipv6_hdr(skb)->saddr,
